pkg/storage/tsdb/config.go

```diff
@@ -452,6 +457,9 @@ func (cfg *BucketStoreConfig) Validate(logger log.Logger) error {
 	if !util.StringsContain(validSeriesSelectionStrategies, cfg.SeriesSelectionStrategyName) {
 		return errors.New("invalid series-selection-strategy, set one of " + strings.Join(validSeriesSelectionStrategies, ", "))
 	}
+	if cfg.SelectionStrategies.WorstCaseSeriesPreference <= 0 {
```

Comment:
[nit] I would enforce it only when the strategy is "worst-case".

Reply:
I did it in https://github.com/grafana/mimir/pull/5047/commits/a924c6da3746cfb4c02e55fafac0436367d077df